### PR TITLE
fix fedora release

### LINKED ISSUE #14902: Experimental script fails on fedora22
If more than one rpm is present which matches che query \*-release the script will generate an invalid yum repo configuration as per lines 141:144

```
         if [ -z "$lsb_dist" ] && [ -r /etc/fedora-release ]; then
                 lsb_dist='fedora'
                 dist_version="$(rpm -qa \*-release | cut -d"-" -f3)"
         fi
```

in my system the rpm command returns the following result
```
$ rpm -qa \*-release
fedora-release-22-1.noarch
rpmfusion-nonfree-release-22-1.noarch
rpmfusion-free-release-22-1.noarch
```

and the script generates this (broken) repo file

```
$ cat /etc/yum.repos.d/docker-experimental.repo
[docker-experimental-repo]
name=Docker experimental Repository
baseurl=https://yum.dockerproject.org/repo/experimental/fedora/22
release
release
enabled=1
gpgcheck=1
gpgkey=https://yum.dockerproject.org/gpg
```

changing the query from \*-release to \*fedora-release solves the problem

```
         if [ -z "$lsb_dist" ] && [ -r /etc/fedora-release ]; then
                 lsb_dist='fedora'
                 dist_version="$(rpm -qa \*fedora-release | cut -d"-" -f3)"
         fi
```

### PR DESCRIPTION
**Pull request by jessfraz**
fixes https://github.com/docker/docker/issues/14902

ping @tianon

### CODE REVIEW

**Comment by jfrazelle** [comment]
actually i dont like this fix im going to do it differently

**Comment by jfrazelle** [comment]
ok changed

File: hack/install.sh [hunk 1]
```diff
@@ -140,12 +140,12 @@ do_install() {
 	fi
 	if [ -z "$lsb_dist" ] && [ -r /etc/fedora-release ]; then
 		lsb_dist='fedora'
-		dist_version="$(rpm -qa \*-release | cut -d"-" -f3)"
+		dist_version="$(rpm -qa \*-release | cut -d"-" -f3 | cut -f1 -d"\n")"
```
Comment (icecrime) [inline hack/install.sh]: Isn't ` | head -n1` better to split on newlines?

Comment (jfrazelle) [inline hack/install.sh]: i debated haha will change

Comment (jfrazelle) [inline hack/install.sh]: updated!

**Comment by tianon** [comment]
LGTM

**Comment by icecrime** [comment]
LGTM

**Comment by Djelibeybi** [comment]
:-1:  This is not a great fix because it still reports on \*-release. Adding the head -n1 just reports on the first one, but that still breaks in certain circumstances, e.g.:

    $ rpm -qa \*-release
    epel-release-6-8.noarch
    oraclelinux-release-6Server-6.0.2.x86_64

Thus, the version returned by ```rpm -qa \*-release | cut -d"-" -f3 | head -n1``` is the version for ```epel-release``` not the right version (```6Server```) from oraclelinux-release.

The more correct way would be to query on the RPM that provides ```redhat-release```, e.g.:

```rpm -q --whatprovides redhat-release```

This should work on Fedora as well as RHEL and its derivatives.

**Comment by jfrazelle** [comment]
pull requests welcome

**Pull request merged**